msticpy/data/drivers/kusto_driver.py

```diff
@@ -19,12 +19,9 @@
 __version__ = VERSION
 __author__ = "Ian Hellen"
 
-_KCS_TEMPLATE = (
-    "azure_data-Explorer://"
-    "tenant='{tenant}';clientid='{clientid}';clientsecret='{clientsecret}';"
-    "cluster='{cluster}';"
-    "database='{database}'"
-)
+_KCS_CODE = "code;"
+_KCS_APP = "tenant='{tenant}';clientid='{clientid}';clientsecret='{clientsecret}';"
+_KCS_TEMPLATE = "azure_data-Explorer://{auth}cluster='{cluster}';database='{database}'"
```

Comment:
Its probably not worth taking a dependency but this could be useful - https://github.com/Azure/azure-kusto-python/blob/35e23cef5d246328245a6e34c642db818aec1b19/azure-kusto-data/azure/kusto/data/client.py

Reply:
This is interesting. I'm thinking of looking at it to potentially replace Kqlmagic.
It had problems initially (like no LogAnalytics) but might have worked through them now